### ISSUE TITLE
Report error when images are out of date with manifest

### ISSUE THREAD

**Issue by sjezewski** [pachyderm/pachyderm#358]
I ran into this issue just now.

The manifest changed (I didn't realize this), and when I was doing `make clean-launch` and `make launch` I kept getting consistent rethink errors on the pods:

```shell
gorethink: Database `pachyderm` does not exist. in: 
r.DB("pachyderm").Table("JobInfos").Wait()
```

But the rethink pod seemed fine:
```shell
Recursively removing directory /var/rethinkdb/data/tmp
Initializing directory /var/rethinkdb/data
Running rethinkdb 2.2.6~0jessie (GCC 4.9.2)...
Running on Linux 4.2.0-35-generic x86_64
Loading data from directory /var/rethinkdb/data
Listening for intracluster connections on port 29015
Listening for client driver connections on port 28015
Listening for administrative HTTP connections on port 8080
Listening on addresses: 127.0.0.1, 172.17.0.6, ::1, fe80::42:acff:fe11:6%124
Server ready, "rethink_7rvpe_khm" ce39a1fb-f3ca-47bd-9618-3abbeec4dc75
A newer version of the RethinkDB server is available: 2.3.1. You can read the changelog at <https://github.com/rethinkdb/rethinkdb/releases>.
```

JD suggested that the manifest changed and that I re-run `make docker-build` --- this fixed the issue.

However, it was not clear at all that's what was happening. This is confusing to users. We can make this better. 

So I'm proposing we:

- bake the checksum of the manifest into the image
- then as part of the `make launch` task, do something like:

```shell
docker run pachd cat some_generated_checksum_of_manifest_that_existed_when_this_image_was_built > existing_manifest_version
cat /etc/compile/manifest.json | sha256 > newest_manifest_version
diff existing_manifest_version newest_manifest_version
if [ -ne 0 `echo $?` ]]; then \
    echo "Unable to launch w outdated images. Please run 'make docker-build' before launching"
    exit 1
fi
```

Basically - check that some file (that we'll have to generate and bake into the image) matches a local checksum of the manifest, and prevent launching w a descriptive error message if there is a mismatch

**Comment by sjezewski** [comment]
Fixed in https://github.com/pachyderm/pachyderm/pull/510
